core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/sendsemantics/partitioners/HashBasedShufflePartitioner.scala

```diff
@@ -8,13 +8,13 @@ case class HashBasedShufflePartitioner(partitioning: HashBasedShufflePartitionin
     extends Partitioner {
 
   override def getBucketIndex(tuple: Tuple): Iterator[Int] = {
-    val numBuckets = partitioning.receivers.length
+    val numBuckets = partitioning.channels.length
     val partialTuple =
       if (partitioning.hashAttributeNames.isEmpty) tuple
       else tuple.getPartialTuple(partitioning.hashAttributeNames.toList)
     val index = Math.floorMod(partialTuple.hashCode(), numBuckets)
     Iterator(index)
   }
 
-  override def allReceivers: Seq[ActorVirtualIdentity] = partitioning.receivers
+  override def allReceivers: Seq[ActorVirtualIdentity] = partitioning.channels.map(_.toWorkerId)
```

Comment:
just want to make sure there are no duplicate receiver ids (in case of having multiple channels to the same receiver worker). You might want to change the `Seq` to `Set` to ensure that.

Reply:
Fixed by using `distinct`